pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -1050,3 +1246,105 @@ func (r *Reconciler) findObjectsForSecret(ctx context.Context, secret client.Obj
 	}
 	return requests
 }
+
+// ensureWatchForGVK ensures that a watch is registered for the given GroupVersionKind.
+// This enables drift detection for non-Secret target resources.
+func (r *Reconciler) ensureWatchForGVK(gvk schema.GroupVersionKind) error {
+	// Check if already watching this GVK
+	if r.watchTracker.IsWatched(gvk) {
+		return nil
+	}
+
+	if r.controller == nil {
+		return fmt.Errorf("controller not initialized, cannot register dynamic watch")
+	}
+
+	// TODO: Separate informer for watches.
+	// Use PartialObjectMetadata to only cache metadata (name, labels, annotations, resourceVersion)
+	// instead of full objects. This reduces memory usage since we refetch full objects during reconciliation anyway.
+	partial := &metav1.PartialObjectMetadata{}
+	partial.SetGroupVersionKind(gvk)
+
+	// predicates for dynamic watches
+	hasManagedLabel := predicate.TypedFuncs[*metav1.PartialObjectMetadata]{
+		CreateFunc: func(e event.TypedCreateEvent[*metav1.PartialObjectMetadata]) bool {
+			value, hasLabel := e.Object.GetLabels()[esv1.LabelManaged]
+			return hasLabel && value == esv1.LabelManagedValue
+		},
+		UpdateFunc: func(e event.TypedUpdateEvent[*metav1.PartialObjectMetadata]) bool {
+			value, hasLabel := e.ObjectNew.GetLabels()[esv1.LabelManaged]
+			return hasLabel && value == esv1.LabelManagedValue
+		},
+		DeleteFunc: func(e event.TypedDeleteEvent[*metav1.PartialObjectMetadata]) bool {
+			value, hasLabel := e.Object.GetLabels()[esv1.LabelManaged]
+			return hasLabel && value == esv1.LabelManagedValue
+		},
+	}
+
+	// Create a Kind source for watching this type
+	// source.Kind creates a source that watches the specified kind
+	// TODO: Ward against overloading the informer cache.
+	src := source.Kind(
+		r.cache,
```

Comment:
I believe this is never stopped if the target is removed. is that really so or am i simply missing something?

Reply:
Hmmmmm I feel like this is a lot more complex than it needs to be. :D But I'll see what I can use out of this.